stdlib/public/Concurrency/Clock.cpp

```diff
@@ -52,14 +56,15 @@ void swift_get_time(
       QueryPerformanceFrequency(&freq);
       LARGE_INTEGER count;
       QueryPerformanceCounter(&count);
-      *seconds = count.QuadPart / freq.QuadPart;
-      if (freq.QuadPart < 1000000000) {
-        *nanoseconds = 
-            ((count.QuadPart % freq.QuadPart) * 1000000000) / freq.QuadPart;
-      } else {
-        *nanoseconds = 
-            (count.QuadPart % freq.QuadPart) * (1000000000.0 / freq.QuadPart);
-      }
+      // Divide count (number of ticks) by frequency (number of ticks per
+      // second) to get the counter in seconds. We also need to multiply the
+      // count by 1,000,000,000 to get nanosecond resolution. By multiplying
+      // first, we maintain high precision. The resulting value is the tick
+      // count in nanoseconds. Use 128-bit math to avoid overflowing.
+      auto quadPart = static_cast<unsigned __int128>(count.QuadPart);
+      auto ns = (quadPart * 1'000'000'000) / freq.QuadPart;
+      *seconds = ns / 1'000'000'000;
+      *nanoseconds = ns % 1'000'000'000;
```

Comment:
Nice :-)

Reply:
Yeah, the "which compiler builds this file?" part tripped me up!